Library/Homebrew/unpack_strategy.rb

```diff
@@ -135,14 +135,27 @@ def initialize(path, ref_type: nil, ref: nil, merge_xattrs: nil)
   def extract_to_dir(unpack_dir, basename:, verbose:); end
   private :extract_to_dir
 
-  def extract(to: nil, basename: nil, verbose: nil)
+  sig {
+    params(
+      to: T.nilable(Pathname), basename: T.nilable(T.any(String, Pathname)), verbose: T::Boolean,
```

Comment:
Not necessarily for this PR, but i wonder if want to create a type alias for this one:
```
$ git grep -o 'T.any(String, Pathname)' | wc -l
56
```

Reply:
@dduugg Good idea!